client/service/src/task_manager/mod.rs

```diff
@@ -124,18 +124,17 @@ impl SpawnTaskHandle {
 			}
 		};
 
-		let future: Pin<Box<dyn Future<Output = ()> + Send>> =
-			if let Some(telemetry_span) = self.telemetry_span.clone() {
-				// this will preserve the telemetry span and by extension all of its parents as they
-				// are linked together upon creation
-				// this is used by sc_telemetry::layer::TelemetryLayer
-				Box::pin(future.instrument(telemetry_span.span()))
-			} else {
-				// this will preserve the current span (if any) and by extension all of its parents
-				// as they are linked together upon creation
-				// this is used by sc_tracing::logging::prefix_logs_with
-				Box::pin(future.in_current_span())
-			};
+		let future: BoxFuture<()> = if let Some(telemetry_span) = self.telemetry_span.clone() {
+			// this will preserve the telemetry span and by extension all of its parents as they
+			// are linked together upon creation
+			// this is used by sc_telemetry::layer::TelemetryLayer
+			Box::pin(future.instrument(telemetry_span.span()))
+		} else {
+			// this will preserve the current span (if any) and by extension all of its parents
+			// as they are linked together upon creation
+			// this is used by sc_tracing::logging::prefix_logs_with
```

Comment:
```suggestion
			// This will preserve the telemetry span and by extension all of its 
			// parents as they are linked together upon creation.
			// Used by [`sc_telemetry::layer::TelemetryLayer`].
			Box::pin(future.instrument(telemetry_span.span()))
		} else {
			// This will preserve the current span (if any) and by extension all 
			// of its parents as they are linked together upon creation.
			// Used by [`sc_tracing::logging::prefix_logs_with`].
```